### PR TITLE
rgw: avoid expiration early triggering caused by overflow

### PR DESCRIPTION
**Pull request by jiahuizeng**
Fixes: https://tracker.ceph.com/issues/42634

Signed-off-by: jiahuizeng <jhzeng93@foxmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -309,11 +309,11 @@ static bool obj_has_expired(CephContext *cct, ceph::real_time mtime, int days, c
   utime_t base_time;
   if (cct->_conf->rgw_lc_debug_interval <= 0) {
     /* Normal case, run properly */
-    cmp = days*24*60*60;
+    cmp = (long)days*24*60*60;
```
Comment (liuchang0812) [inline src/rgw/rgw_lc.cc]: `cmp = (double)days*24*60*60;` or ` cmp = days*24ll*60*60;`

**Comment by liuchang0812** [comment]
@cbodley mind taking a look?

**Comment by liuchang0812** [comment]
ping

File: src/rgw/rgw_lc.cc [hunk 2]
```diff
@@ -309,11 +309,11 @@ static bool obj_has_expired(CephContext *cct, ceph::real_time mtime, int days, c
   utime_t base_time;
   if (cct->_conf->rgw_lc_debug_interval <= 0) {
     /* Normal case, run properly */
-    cmp = days*24*60*60;
+    cmp = (double)days*24*60*60;
```
Comment (mattbenjamin) [inline src/rgw/rgw_lc.cc]: prefer double(expr), but this is portable C

**Comment by mattbenjamin** [comment]
thanks @jiahuizeng, looks good

**Comment by theanalyst** [comment]
needs rebase

**Comment by jiahuizeng** [comment]
> needs rebase

@theanalyst resolved

**Comment by theanalyst** [comment]
jenkins test this please

**Comment by ivancich** [comment]
jenkins test make check

**Comment by ivancich** [comment]
Teuthology runs:

- rgw/verify: http://pulpito.ceph.com/ivancich-2020-12-01_16:21:42-rgw-wip-eric-testing-1-distro-basic-smithi/
- rgw/(non-)verify: http://pulpito.ceph.com/ivancich-2020-12-01_16:21:59-rgw-wip-eric-testing-1-distro-basic-smithi/

**Comment by ivancich** [comment]
@jiahuizeng It seems like this bug fix needs to be backported, but the tracker did not list any backports. I added them, but in the future, please make sure the tracker lists backports. Thanks!

**Comment by mattbenjamin** [comment]
thanks, @ivancich

**Pull request merged**